### DISCUSSION TITLE
How to use AWS IoT Device SDK for Embedded C with Micropython

### DISCUSSION THREAD

**Discussion by kulmam92** [micropython/micropython#9239] [Libraries & Drivers] [answered]
There's no official aws iot device sdk for micropython. But is there a way to use AWS IoT Device SDK for Embedded C with Micropython?

The followings are the related links that I was able to find but none of them provides any solutions.
[Embed aws iot device sdk to Micropython as user C module - stackoverflow question](https://stackoverflow.com/questions/64157384/embed-aws-iot-device-sdk-to-micropython-as-user-c-module)
[Releasing LTS of AWS IoT Device SDK for Embedded C on ESP32](https://blog.espressif.com/support-for-lts-release-of-aws-iot-device-sdk-for-embedded-c-on-esp32-8eeeea28b79b)

**Comment by jimmo** [comment, marked as answer]
> But is there a way to use AWS IoT Device SDK for Embedded C with Micropython?

This will likely be quite difficult as they will probably expect C APIs for networking and sockets to be available. (i.e. it's probably expecting to run on FreeRTOS).

It will probably be less work to try and replicate the basics of what you need from the SDK in Python.
